fastcrypto-zkp/src/conversions.rs

```diff
@@ -176,6 +199,16 @@ pub fn bls_g2_affine_to_blst_g2_affine(pt: &BlsG2Affine) -> blst_p2_affine {
     unsafe {
         blst_p2_affine_serialize(tmp2.as_mut_ptr(), &tmp_p2);
     };
+    // BLST disagrees with Arkworks on the uncompressed representation of the infinity point on G2
+    // https://github.com/supranational/blst/blob/6382d67c72119d563975892ed49ba32e92d3d0da/src/e2.c#L194-L203
+    // the infinity bit notwithstanding,
+    // Arkworks: x = Fq2::zero(), y = Fq2::one()
+    // BLST: x = y = Fq2::zero()
+    // BLST follows the standard here (see https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#appendix-C)
+    if pt.infinity {
+        tmp2[0] = 0x40; // set the infinity bit
+        tmp2[96..].copy_from_slice(&[0u8; 96]); // zeroize the rest of the point
```

Comment:
ditto, I think you don't need to copy the full slice but just set one bit to convert `Fq2::one()` -> `Fq2::zero()` (all zero bytes)

Reply:
Same as [the above](https://github.com/MystenLabs/fastcrypto/pull/149#discussion_r1003412930): I think it's safer to not assume the caller will use Arkworks conventions, and return the correct result over the whole input range.

Reply:
another option is to set this a const and just use that. IMO that would safer vs having code to reconstruct the infinity point every single time.

Reply:
Right, so given the `blst` definitions all call into FFI functions (all non-const), I don't have the ability to make a const definition, here. But I can certainly make auxiliary functions for clarity.